### PR TITLE
hack/tools: Update zeitgeist to v0.2.0 to drop import cycle

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this?

/kind cleanup
/area dependency

#### What this PR does / why we need it:

zeitgeist, the underlying tool for `hack/verify-external-dependencies.sh` was pulling in a ton of transitive dependencies (including k8s.io/kubernetes@v1.13.0) due to a few things that have been fixed with https://github.com/kubernetes-sigs/zeitgeist/releases/tag/v0.2.0.

https://github.com/kubernetes/kubernetes/pull/98845#discussion_r584370625, https://github.com/kubernetes-sigs/zeitgeist/pull/32

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

This also addresses superfluous dependency log warnings via https://github.com/kubernetes-sigs/zeitgeist/pull/33.
ref: https://kubernetes.slack.com/archives/C2C40FMNF/p1614284822055200, https://github.com/kubernetes/kubernetes/pull/98946#issuecomment-786177368, https://github.com/kubernetes/kubernetes/pull/99637
cc: @monopole @pacoxu  

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by pacoxu** [comment]
👍
/lgtm

**Comment by justaugustus** [comment]
Not quite ready yet, but just FYI @dims @liggitt @BenTheElder.

File: hack/tools/go.sum [hunk 1]
```diff
@@ -1672,78 +909,16 @@ honnef.co/go/tools v0.0.1-2020.1.3/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9
...
-k8s.io/klog v0.3.0/go.mod h1:Gq+BEi5rUBO/HRz0bTSXDUcqjScdoY3a9IHpCEIOOfk=
-k8s.io/klog v1.0.0 h1:Pt+yjF5aB1xDSVbau4VsWe+dQNzA0qv1LlXdC2dF6Q8=
-k8s.io/klog v1.0.0/go.mod h1:4Bi6QPql/J/LkTDqv7R/cd3hPo4k2DG6Ptcz060Ez5I=
-k8s.io/klog/v2 v2.0.0/go.mod h1:PBfzABfn139FHAV07az/IF9Wp1bkk3vpT2XSJ76fSDE=
-k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a/go.mod h1:1TqjTSzOxsLGIKfj0lK8EeCP7K1iUG65v09OM0/WG5E=
-k8s.io/kube-openapi v0.0.0-20200410145947-61e04a5be9a6 h1:Oh3Mzx5pJ+yIumsAD0MOECPVeXsVot0UkiaCGVyfGQY=
-k8s.io/kube-openapi v0.0.0-20200410145947-61e04a5be9a6/go.mod h1:GRQhZsXIAJ1xR0C9bd8UpWHZ5plfAS9fzPjJuQ6JL3E=
-k8s.io/kubectl v0.18.8/go.mod h1:PlEgIAjOMua4hDFTEkVf+W5M0asHUKfE4y7VDZkpLHM=
-k8s.io/kubernetes v1.13.0 h1:qTfB+u5M92k2fCCCVP2iuhgwwSOv1EkAkvQY1tQODD8=
-k8s.io/kubernetes v1.13.0/go.mod h1:ocZa8+6APFNC2tX1DZASIbocyYT5jHzqFVsY5aoB7Jk=
```
Comment (justaugustus) [inline hack/tools/go.sum:1703]: Ooooo, shiny.

**Comment by justaugustus** [comment]
/assign @spiffxp @BenTheElder @dims 
cc: @kubernetes/release-engineering

**Review by cpanato — APPROVED** [review]
/lgtm

**Review by xmudrii — APPROVED** [review]
/lgtm

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by liggitt** [comment]
nice... cut the time for this verify task by more than 50%:

```
verify-external-dependencies-version.sh	47s
```

before this:
```
verify-external-dependencies-version.sh	102s
```

**Comment by BenTheElder** [comment]
Thank you!

**Pull request merged**